test/integration/Pool/pool.js

```diff
@@ -49,50 +50,46 @@ async function tokenPriceSetup() {
-  //   const daiTokenPrice = await pool.getInternalTokenPriceInAsset(1);
-  //
-  //   const totalAssetValue = await pool.getPoolValueInEth();
-  //   const mcrEth = await mcr.getMCR();
-  //   const expectedEthTokenPrice = BigNumber.from(getTokenSpotPrice(totalAssetValue, mcrEth).toString());
-  //
-  //   const ethPriceDiff = ethTokenPrice.sub(expectedEthTokenPrice).abs();
-  //   assert(
-  //     ethPriceDiff.lte(BigNumber.from(1)),
-  //     `token price ${ethTokenPrice.toString()} not close enough to ${expectedEthTokenPrice.toString()}`,
-  //   );
-  //
-  //   const expectedDaiPrice = BigNumber.from(ethToDaiRate / 100).mul(expectedEthTokenPrice);
-  //   const daiPriceDiff = daiTokenPrice.sub(expectedDaiPrice);
-  //   assert(
-  //     daiPriceDiff.lte(BigNumber.from(10000)), // negligible amount of wei
-  //     `DAI token price ${daiTokenPrice.toString()} not close enough to ${expectedDaiPrice.toString()}`,
-  //   );
-  // });
+  it('getInternalTokenPriceInAsset returns spot price for all assets', async function () {
+    const fixture = await loadFixture(tokenPriceSetup);
+    const { p1: pool, mcr, tc, ra } = fixture.contracts;
+    const { ethToDaiRate } = fixture.rates;
+
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const expectedTokenPriceInEth = await getInternalPrice(ra, pool, tc, mcr, timestamp);
+    const expectedTokenPriceInDai = BigNumber.from(ethToDaiRate / 100).mul(expectedTokenPriceInEth);
+
+    const ethTokenPrice = await pool.getInternalTokenPriceInAsset(PoolAsset.ETH);
+    const daiTokenPrice = await pool.getInternalTokenPriceInAsset(PoolAsset.DAI);
+
+    expect(ethTokenPrice).to.be.equal(expectedTokenPriceInEth);
+    expect(daiTokenPrice).to.be.equal(expectedTokenPriceInDai);
+  });
 
   it('getPoolValueInEth calculates pool value correctly', async function () {
     const fixture = await loadFixture(tokenPriceSetup);
-    const { p1: pool, dai } = fixture.contracts;
-    const { daiToEthRate } = fixture.rates;
+    const { p1: pool, dai, usdc } = fixture.contracts;
+    const { daiToEthRate, usdcToEthRate } = fixture.rates;
+    const { USDC_DECIMALS } = fixture.config;
+
+    const totalAssetValue = await pool.getPoolValueInEth();
 
     const poolBalance = BigNumber.from(await ethers.provider.getBalance(pool.address));
+
     const daiBalance = await dai.balanceOf(pool.address);
-    const expectedDAiValueInEth = daiToEthRate.mul(daiBalance).div(parseEther('1'));
-    const expectedTotalAssetValue = poolBalance.add(expectedDAiValueInEth);
-    const totalAssetValue = await pool.getPoolValueInEth();
-    assert(totalAssetValue.toString(), expectedTotalAssetValue.toString());
+    const expectedDaiValueInEth = daiToEthRate.mul(daiBalance).div(parseEther('1'));
+
+    const usdcBalance = await usdc.balanceOf(pool.address);
+    const expectedUsdcValueInEth = usdcToEthRate.mul(usdcBalance).div(parseUnits('1', USDC_DECIMALS));
```

Comment:
The assertion for this test was previously wrong so it was always passing even though it was not taking into account the USDC balance.

```javascript
assert(totalAssetValue.toString(), expectedTotalAssetValue.toString()); // wrong assertion
```